hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/AbstractContainerReportHandler.java

```diff
@@ -376,6 +377,33 @@ private boolean updateContainerState(final DatanodeDetails datanode,
     return ignored;
   }
 
+  /**
+   * Helper method to verify that the replica's bcsId matches the container's in SCM.
+   * Throws IOException if the bcsIds do not match.
+   * <p>
+   * @param replicaBcsId Replica bcsId
+   * @param containerBcsId Container bcsId in SCM
+   * @param datanode DatanodeDetails for logging
+   * @param containerId ContainerID for logging
+   * @return true if verification has passed, false otherwise
+   * @throws IOException Thrown when bcsIds do not match
```

Comment:
The method doesn't throw `IOException` if the bcsIds do not match.

Reply:
done